# Prepare data

## Load data

Load the original data files outputted by the bioinformatic pipeline.

```{r load_read_counts, warning=FALSE, comments="", message=FALSE}
read_counts <- read_tsv("resources/coverm/genome.count.tsv.xz") %>%
    rename(genome=1)
```

```{r load_genome_hits, warning=FALSE, comments="", message=FALSE}
genome_hits <- read_tsv("resources/coverm/genome.covered_bases.tsv.xz") %>%
    rename(genome=1)
```

```{r load_genome_taxonomy, warning=FALSE, comments="", message=FALSE}
genome_taxonomy <- read_tsv("resources/gtdbtk/gtdbtk.summary.tsv.xz") %>%
  select(mag_id = user_genome, classification) %>%
  separate(
    classification,
    into = c("domain", "phylum", "class", "order", "family", "genus", "species"),
    sep = ";") %>%
    rename(genome=1)
```

```{r load_genome_quality, warning=FALSE, comments="", message=FALSE}
genome_quality <- read_tsv("resources/checkm2/quality_report.tsv.xz") %>%
  select(
    genome = Name, 
    completeness = Completeness, 
    contamination = Contamination,
    length = Genome_Size, 
    gc = GC_Content
  )
```

```{r load_genome_tree, warning=FALSE, comments="", message=FALSE}
genome_tree <- read_tree("resources/gtdbtk/gtdbtk.backbone.bac120.classify.tree")
genome_tree$tip.label <- str_replace_all(genome_tree$tip.label,"'", "") #remove single quotes in MAG names
genome_tree <- keep.tip(genome_tree, tip=genome_taxonomy$genome) # keep only MAG tips
```

```{r load_sample_metadata, warning=FALSE, comments="", message=FALSE}
sample_metadata <- read_tsv("resources/metadata.tsv")
```

## Create working objects

Transform the original data files into working objects for downstream analyses.

```{r generate_genome_metadata, warning=FALSE, comments="", message=FALSE}
genome_metadata <- genome_taxonomy %>%
  left_join(genome_quality,by=join_by(genome==genome)) #join quality
```

```{r calculate_genome_coverage, warning=FALSE, comments="", message=FALSE}
genome_coverage <- genome_hits %>%
  mutate(across(where(is.numeric), ~ ./genome_metadata$length))
```

```{r filter_coverage, warning=FALSE, comments="", message=FALSE}
min_coverage=0.3
read_counts_filt <- genome_coverage %>%
  mutate(across(where(is.numeric), ~ ifelse(. > min_coverage, 1, 0))) %>%
  mutate(across(-1, ~ . * read_counts[[cur_column()]])) 
```

```{r calculate_genome_counts_unfiltered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts <- read_counts %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

```{r calculate_genome_counts_filtered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts_filt <- read_counts_filt %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

## Prepare color scheme

[AlberdiLab](www.alberdilab.dk) projects use unified color schemes developed for the [Earth Hologenome Initiative](www.earthhologenome.org), to facilitate figure interpretation.

```{r get_ehi_colors, warning=FALSE, comments="", message=FALSE}
phylum_colors <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(phylum, colors) %>% 
    unique() %>%
    arrange(phylum) %>%
    select(colors) %>%
    pull()
```

## Wrap working objects

All working objects are wrapped into a single Rdata object to facilitate downstream usage.

```{r wrap_working_objects, warning=FALSE, comments="", message=FALSE}
save(read_counts, genome_counts, genome_counts_filt, genome_tree, genome_metadata, sample_metadata, phylum_colors, file = "resources/data.Rdata")
```
