# Prepare data

## Load data

Load the original data files outputted by the bioinformatic pipeline.

### Sample metadata

```{r load_sample_metadata, warning=FALSE, comments="", message=FALSE}
sample_metadata <- read_tsv("resources/metadata.tsv")
```

### Read counts

```{r load_read_counts, warning=FALSE, comments="", message=FALSE}
read_counts <- read_tsv("resources/coverm/genome.count.tsv.xz") %>%
    rename(genome=1)
```

### Genome base hits

```{r load_genome_hits, warning=FALSE, comments="", message=FALSE}
genome_hits <- read_tsv("resources/coverm/genome.covered_bases.tsv.xz") %>%
    rename(genome=1)
```

### Genome taxonomy

```{r load_genome_taxonomy, warning=FALSE, comments="", message=FALSE}
genome_taxonomy <- read_tsv("resources/gtdbtk/gtdbtk.summary.tsv.xz") %>%
  select(mag_id = user_genome, classification) %>%
  separate(
    classification,
    into = c("domain", "phylum", "class", "order", "family", "genus", "species"),
    sep = ";") %>%
    rename(genome=1)
```

### Genome quality

```{r load_genome_quality, warning=FALSE, comments="", message=FALSE}
genome_quality <- read_tsv("resources/checkm2/quality_report.tsv.xz") %>%
  select(
    genome = Name, 
    completeness = Completeness, 
    contamination = Contamination,
    length = Genome_Size, 
    gc = GC_Content
  )
```

### Genome tree

```{r load_genome_tree, warning=FALSE, comments="", message=FALSE}
genome_tree <- read_tree("resources/gtdbtk/gtdbtk.backbone.bac120.classify.tree")
genome_tree$tip.label <- str_replace_all(genome_tree$tip.label,"'", "") #remove single quotes in MAG names
genome_tree <- keep.tip(genome_tree, tip=genome_taxonomy$genome) # keep only MAG tips
```

### Genome annotations

```{r load_genome_annotations, warning=FALSE, comments="", message=FALSE}
genome_annotations <- read_tsv("resources/dram/annotations.tsv.xz") %>%
    rename(gene=1, genome=2, contig=3)
```

## Create working objects

Transform the original data files into working objects for downstream analyses.

### Merge genome taxonomy and quality

```{r generate_genome_metadata, warning=FALSE, comments="", message=FALSE}
genome_metadata <- genome_taxonomy %>%
  left_join(genome_quality,by=join_by(genome==genome)) #join quality
```

### Calculate genome coverage

```{r calculate_genome_coverage, warning=FALSE, comments="", message=FALSE}
genome_coverage <- genome_hits %>%
  mutate(across(where(is.numeric), ~ ./genome_metadata$length))
```

### Filter reads by coverage

```{r filter_coverage, warning=FALSE, comments="", message=FALSE}
min_coverage=0.3
read_counts_filt <- genome_coverage %>%
  mutate(across(where(is.numeric), ~ ifelse(. > min_coverage, 1, 0))) %>%
  mutate(across(-1, ~ . * read_counts[[cur_column()]])) 
```

### Transform reads into genome counts

```{r calculate_genome_counts_unfiltered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts <- read_counts %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

```{r calculate_genome_counts_filtered, warning=FALSE, comments="", message=FALSE}
readlength=150
genome_counts_filt <- read_counts_filt %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$length / readlength) ))
```

### Distill annotations into GIFTs 

```{r distill_annotations, warning=FALSE, comments="", message=FALSE}
genome_gifts <- distill(genome_annotations,GIFT_db,genomecol=2,annotcol=c(9,10,19), verbosity=F)
```

## Load data statistics

### Raw reads

```{r raw_reads, warning=FALSE, comments="", message=FALSE}
raw_reads <-
  "resources/report/by_step/reads_data/multiqc_general_stats.txt.xz" %>%
  read_tsv() %>%
  select(
    sample = Sample,
    raw_reads = `FastQC_mqc-generalstats-fastqc-total_sequences`
  ) %>%
  mutate(
    sample = sample %>% str_remove_all("_1$") %>% str_remove_all("_2$")
  ) %>%
  summarise(raw_reads = sum(raw_reads), .by = sample)
```

### Quality-filtered reads

```{r filtered_reads, warning=FALSE, comments="", message=FALSE}
fastp_reads <-
  "resources/report/by_step/preprocessing_data/multiqc_general_stats.txt.xz" %>%
  read_tsv() %>%
  filter(str_detect(Sample, "fastp")) %>%
  select(
    sample = Sample,
    trimmed_reads = `FastQC_mqc-generalstats-fastqc-total_sequences`
  ) %>%
  mutate(
    sample =
      sample %>%
        str_remove_all("_[u12]+$") %>%
        str_remove_all("^fastp \\| ")
  ) %>%
  summarise(trimmed_reads = sum(trimmed_reads), .by = sample)
```

### Host-mapped reads

```{r host_mapped, warning=FALSE, comments="", message=FALSE}
host_mapped <-
  "resources/report/by_step/preprocessing_data/multiqc_general_stats.txt.xz" %>%
  read_tsv() %>%
  filter(!str_detect(Sample, "fastp")) %>%
  select(
    sample = Sample,
    host_mapped = `Samtools_mqc-generalstats-samtools-reads_mapped`,
    mapping_total = `Samtools_mqc-generalstats-samtools-raw_total_sequences`
  ) %>%
  mutate(
    host_unmapped = mapping_total - host_mapped
  ) %>%
  filter(!is.na(host_mapped)) %>%
  separate(
    col = sample,
    into = c("host_name", "sample"),
    sep = " \\| "
  ) %>%
  rename(mapped = host_mapped, unmapped = host_unmapped) %>%
  select(-mapping_total) %>%
  pivot_longer(-host_name:-sample) %>%
  mutate(
    name = str_glue("{name}_{host_name}")
  ) %>%
  select(-host_name) %>%
  pivot_wider()
```

### Prokaryotic fraction

```{r singlem, warning=FALSE, comments="", message=FALSE}
singlem <-
  "resources/singlem/microbial_fraction.tsv.xz" %>%
  read_tsv() %>%
  distinct() %>%
  mutate(
    sample = sample %>% str_remove_all("_1$"),
    read_fraction = read_fraction %>% str_remove("%") %>% as.numeric(),
    read_fraction = read_fraction / 100
  ) %>%
  select(
    sample,
    singlem_prokaryotic_bases = bacterial_archaeal_bases,
    singlem_metagenome_size = metagenome_size,
    singlem_read_fraction = read_fraction,
  )
```

### Metagenomic complexity

```{r nonpareil, warning=FALSE, comments="", message=FALSE}
nonpareil <-
  "resources/nonpareil/nonpareil.tsv.xz" %>%
  read_tsv() %>%
  select(sample=sample_id, nonpareil_c = C, nonpareil_diversity = diversity)
```

### MAG-mapped reads

```{r mag_mapping, warning=FALSE, comments="", message=FALSE}
mag_mapping <-
  "resources/coverm/contig.count.tsv.xz" %>%
  read_tsv() %>%
  pivot_longer(-sequence_id) %>%
  summarise(value = sum(value), .by = "name") %>%
  rename(sample = name, mapped_mags = value)
```

### Wrap data statistics

```{r wrap_statistics, warning=FALSE, comments="", message=FALSE}
data_stats <- raw_reads %>%
  left_join(fastp_reads) %>%
  left_join(host_mapped) %>%
  left_join(singlem) %>%
  left_join(nonpareil) %>%
  left_join(mag_mapping)
```

## Prepare color scheme

[AlberdiLab](www.alberdilab.dk) projects use unified color schemes developed for the [Earth Hologenome Initiative](www.earthhologenome.org), to facilitate figure interpretation.

```{r get_ehi_colors, warning=FALSE, comments="", message=FALSE}
phylum_colors <- read_tsv("https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv") %>%
    right_join(genome_metadata, by=join_by(phylum == phylum)) %>%
    arrange(match(genome, genome_tree$tip.label)) %>%
    select(phylum, colors) %>% 
    unique() %>%
    arrange(phylum) %>%
    pull(colors, name=phylum)
```

## Wrap working objects

All working objects are wrapped into a single Rdata object to facilitate downstream usage.

```{r wrap_working_objects, warning=FALSE, comments="", message=FALSE}
save(sample_metadata, 
     genome_metadata, 
     read_counts, 
     genome_counts, 
     genome_counts_filt, 
     genome_tree,
     genome_gifts, 
     phylum_colors,
     data_stats,
     file = "resources/data.Rdata")
```
