---
title: "Lizards sample types data"
author: "Ostaizka Aizpurua"
output:
  html_document: default
  pdf_document: default
date: "2024-01-19"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r libraries, message=FALSE, warning=FALSE, echo=FALSE}
## Load required libraries
library(tidyverse)
library(ape)
library(devtools)
library(ggplot2)
library(hillR)
library(spaa)
library(vegan)
library(hilldiv)
library(phyloseq)
library(phytools)
library(microbiome)
library(matrixStats)
library(microbiomeutilities)
library(lme4)
library(MuMIn)
library(nlme)
library(knitr)
# library(kableExtra)
library(pairwiseAdonis)
library(sjPlot)
library(distillR)
library(RColorBrewer)
library(reshape2)
library(ggpubr)
library(ggdendro)
library(grid)
library(gplots)
library(dendextend)
library(stringr)
library(Rtsne)
library(glue)
library(ggtree)
library(ggnewscale)
library(ggtreeExtra)
library(gridExtra)
library(hilldiv2)
```

```{r directories, comment="", echo=FALSE, message=FALSE, warning=FALSE}
set.seed(0024)
```

```{r}
# Counts table
read_counts <- read_tsv("https://sid.erda.dk/share_redirect/dgIbN9vIM4/results/dereplicate/coverm/genome.count.tsv") %>%
  dplyr::rename(genome = 1)
genomes <- read_counts$genome # create list of genome names
```

```{r}
# Mags taxonomy
genome_taxonomy <- read_tsv("https://sid.erda.dk/share_redirect/dgIbN9vIM4/results/dereplicate/gtdbtk/gtdbtk.summary.tsv") %>%
  dplyr::rename(genome = 1) %>%
  separate(classification, c("domain", "phylum", "class", "order", "family", "genus", "species"), sep = ";") %>%
  select(genome, domain, phylum, class, order, family, genus, species) %>%
  arrange(match(genome, genomes)) %>%
  mutate_at(vars(genome, domain, phylum, class, order, family, genus, species), ~ str_replace(., "[dpcofgs]__", ""))
```

```{r}
# Genome quality
genome_quality <- read_tsv("https://sid.erda.dk/share_redirect/dgIbN9vIM4/results/dereplicate/checkm2/quality_report.tsv") %>%
  dplyr::rename(genome = 1) %>%
  filter(genome %in% genomes) %>%
  arrange(match(genome, genomes))
```

```{r}
# Merged metadata object
# Merge taxonomy, length and quality information
genome_metadata <- genome_taxonomy %>%
  left_join(genome_quality, by = join_by(genome == genome)) # join quality
```

```{r}
# Genome covered bases
basehits <- read_tsv("https://sid.erda.dk/share_redirect/dgIbN9vIM4/results/dereplicate/coverm/genome.covered_bases.tsv") %>%
  dplyr::rename(genome = 1) %>%
  arrange(match(genome, genomes))
# Generate coverage table
genome_coverage <- basehits %>%
  mutate(across(where(is.numeric), ~ . / genome_metadata$Genome_Size))
```

```{r}
# Samples metadata
sample_metadata <- read_csv("https://sid.erda.dk/share_redirect/dgIbN9vIM4/results/dereplicate/coverm/metadata.csv")
metarow <- column_to_rownames(sample_metadata, "sample")
```

```{r}
# Phylogenetic tree
bacteria_tree <- read.tree("https://sid.erda.dk/share_redirect/dgIbN9vIM4/results/dereplicate/gtdbtk/classify/gtdbtk.backbone.bac120.classify.tree")
bacteria_tree$tip.label <- str_replace_all(bacteria_tree$tip.label, "'", "") # remove .fa extension
bacteria_tree <- keep.tip(bacteria_tree, tip = genomes) # keep only MAG tips
```

```{r genome_coverage, comment="", echo=FALSE, message=FALSE, warning=FALSE}
## Coverage filtering and normalisation
read_counts_row <- column_to_rownames(read_counts, "genome")
min_genome_coverage <- 0.3
read_counts_coverage <- genome_coverage %>%
  mutate(across(where(is.numeric), ~ ifelse(. > min_genome_coverage, 1, 0))) %>%
  mutate(across(-1, ~ . * read_counts[[cur_column()]]))

readlength <- 150 # change if sequencing read length is different
genome_counts <- read_counts_coverage %>%
  mutate(across(where(is.numeric), ~ . / (genome_metadata$Genome_Size / readlength)))

genome_counts <- genome_counts %>%
  column_to_rownames(., "genome") %>%
  select(which(!colSums(., na.rm = TRUE) %in% 0)) %>%
  rownames_to_column(., "genome")

genome_counts_rel <- genome_counts %>%
  mutate_at(vars(-genome), ~ . / sum(.))
```

```{r loaddata, comment="", echo=FALSE, message=FALSE, warning=FALSE}
# Genome functions
genome_annotations <- read_tsv("https://sid.erda.dk/share_redirect/dgIbN9vIM4/results/dereplicate/dram/annotations.tsv") %>%
  rename(gene = 1, genome = 2)
```

```{r}
# Genome functions distillation
genome_gifts <- distill(genome_annotations, GIFT_db, genomecol = 2, annotcol = c(9, 10, 19))
```


```{r}
save(read_counts, read_counts_coverage, genome_counts, bacteria_tree, genome_metadata, genome_gifts, sample_metadata, file = "/Users/jtk712/Documents/Projects/lizard_sample_types/resources/data/data.Rdata")
```
